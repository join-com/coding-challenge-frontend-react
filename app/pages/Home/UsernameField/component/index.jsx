import React from 'react';
import PropTypes from 'prop-types';
import { FormattedMessage } from 'react-intl';

import messages from './messages';
import styles from './styles.scss';

const UsernameField = ({onChange, value}) => (
  <label htmlFor="username">
    <FormattedMessage {...messages.showRepositories} />
    <span className={styles.prefix}><FormattedMessage {...messages.atPrefix} /></span>
    <FormattedMessage {...messages.username}>
      {(placeholder) => (
        <input
          className={styles.input}
          id="username"
          type="text"
          placeholder={placeholder}
          value={value}
          onChange={onChange}
        />
      )}
    </FormattedMessage>
  </label>
);

UsernameField.propTypes = {
  onChange: PropTypes.func,
  value: PropTypes.string
};
/* istanbul ignore next*/
UsernameField.defaultProps = {
  onChange: () => {}
};

export default UsernameField;
