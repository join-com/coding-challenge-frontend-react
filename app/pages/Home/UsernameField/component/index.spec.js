import React from 'react';
import renderer from 'react-test-renderer';

import { IntlProvider } from 'react-intl';

import UsernameField from './';

describe('<UsernameField />', () => {
  it('renders an empty username field with a placeholder', () => {
    expect(
      renderer.create(
        <IntlProvider locale="en">
          <UsernameField description="empty field" prefix="@" />
        </IntlProvider>
      ).toJSON()
    ).toMatchSnapshot();
  });

  it('renders a username field with a value set', () => {
    expect(
      renderer.create(
        <IntlProvider locale="en">
          <UsernameField description="not empty field" prefix="@" value="name" />
        </IntlProvider>
      ).toJSON()
    ).toMatchSnapshot();
  });
});
