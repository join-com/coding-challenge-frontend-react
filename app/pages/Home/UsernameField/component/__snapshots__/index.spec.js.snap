// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<UsernameField /> renders a username field with a value set 1`] = `
<label
  htmlFor="username"
>
  <span>
    Show Github repositories by
  </span>
  <span
    className="prefix"
  >
    <span>
      @
    </span>
  </span>
  <input
    className="input"
    id="username"
    onChange={[Function]}
    placeholder="username"
    type="text"
    value="name"
  />
</label>
`;

exports[`<UsernameField /> renders an empty username field with a placeholder 1`] = `
<label
  htmlFor="username"
>
  <span>
    Show Github repositories by
  </span>
  <span
    className="prefix"
  >
    <span>
      @
    </span>
  </span>
  <input
    className="input"
    id="username"
    onChange={[Function]}
    placeholder="username"
    type="text"
    value={undefined}
  />
</label>
`;
